### PR TITLE
Feat/add-multi-cluster-support

### PR DESCRIPTION
**Pull request by chintansakhiya**
## Fixes Issue
#2 

## Changes proposed
- Create a command to get the path of the clusterconfig file, which contains the cluster-name and path of kubeconfig.
- Create reverse proxy for every cluster.
- Separate cluster from url. Eg. `https://127.0.0.1:6443/<cluster-name>/api/v1`
- Forward request to appropriate cluster.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by munir131** [comment]
Typo in title and description of PR

File: cmd/app/options/app.go [hunk 1]
```diff
@@ -81,3 +87,8 @@ func (e *ExtraHeaderOptions) AddFlags(fs *pflag.FlagSet) {
 			"proxied requests as part of the impersonated request. A single key can "+
 			"hold multiple values.")
 }
+
+func (k *Cluster) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&k.Config, "clusters-config", k.Config,
+		"Path to the YAML file containing an array of clusters. Each cluster in the array includes two fields: name and kubeconfig.")
```
Comment (munir131) [inline cmd/app/options/app.go:93]: @chintansakhiya Will that yaml file contain kubeconfig or path of kubeconfig?

File: demo.clusterConfig.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+clusters:
```
Comment (munir131) [inline demo.clusterConfig.yaml:1]: @chintansakhiya change file name to clusterConfig.yaml.example for consistency

File: pkg/proxy/proxy.go [hunk 3]
```diff
@@ -132,66 +138,83 @@ func New(restConfig *rest.Config,
...
+	waitCh, listenerStoppedCh, err := p.serve(http.HandlerFunc(p.httpHandler), stopCh)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return waitCh, listenerStoppedCh, nil
 }
 
+func (p *Proxy) httpHandler(w http.ResponseWriter, r *http.Request) {
+	clusterName := strings.Split(r.URL.Path, "/")[1]
```
Comment (munir131) [inline pkg/proxy/proxy.go:208]: @chintansakhiya Good to check length before accessing non zero element.

**Comment by munir131** [comment]
@chintansakhiya I haven't found role, permission mapping. Is it in separate PR?

File: cmd/app/options/app.go (hunk 1 above)
Comment (chintansakhiya) [reply cmd/app/options/app.go:93]: yaml file contain path of kubeconfig

**Comment by chintansakhiya** [comment]
@munir131 role, permission mapping is in  separate PR

**Pull request merged**